public/javascripts/script.js

```diff
@@ -1,4889 +0,0 @@
-  "yahoo.es",
-  "yahoo.fi",
-  "yahoo.fr",
-  "yahoo.gr",
-  "yahoo.hu",
-  "yahoo.ie",
-  "yahoo.in",
-  "yahoo.it",
-  "yahoo.jp",
-  "yahoo.nl",
-  "yahoo.no",
-  "yahoo.pl",
-  "yahoo.pt",
-  "yahoo.ro",
-  "yahoo.ru",
-  "yahoo.se",
-  "yahoofs.com",
-  "yalla.com",
-  "yalla.com.lb",
-  "yalook.com",
-  "yam.com",
-  "yandex.com",
-  "yandex.pl",
-  "yandex.ru",
-  "yandex.ua",
-  "yapost.com",
-  "yapped.net",
-  "yawmail.com",
-  "yeah.net",
-  "yebox.com",
-  "yehey.com",
-  "yemenmail.com",
-  "yepmail.net",
-  "yert.ye.vc",
-  "yesey.net",
-  "yifan.net",
-  "ymail.com",
-  "yogotemail.com",
-  "yomail.info",
-  "yopmail.com",
-  "yopmail.pp.ua",
-  "yopolis.com",
-  "yopweb.com",
-  "youareadork.com",
-  "youmailr.com",
-  "your-house.com",
-  "your-mail.com",
-  "yourinbox.com",
-  "yourlifesucks.cu.cc",
-  "yourlover.net",
-  "yourname.freeservers.com",
-  "yournightmare.com",
-  "yours.com",
-  "yourssincerely.com",
-  "yoursubdomain.zzn.com",
-  "yourteacher.net",
-  "yourwap.com",
-  "yuuhuu.net",
-  "yyhmail.com",
-  "z1p.biz",
-  "za.com",
-  "zahadum.com",
-  "zaktouni.fr",
-  "zeepost.nl",
-  "zetmail.com",
-  "zhaowei.net",
-  "zhouemail.510520.org",
-  "ziggo.nl",
-  "zionweb.org",
-  "zip.net",
-  "zipido.com",
-  "ziplip.com",
-  "zipmail.com",
-  "zipmail.com.br",
-  "zipmax.com",
-  "zmail.ru",
-  "zoemail.com",
-  "zoemail.org",
-  "zoho.com",
-  "zohomail.com",
-  "zomg.info",
-  "zonnet.nl",
-  "zoominternet.net",
-  "zubee.com",
-  "zuvio.com",
-  "zuzzurello.com",
-  "zwallet.com",
-  "zweb.in",
-  "zxcv.com",
-  "zxcvbnm.com",
-  "zybermail.com",
-  "zydecofan.com",
-  "zzn.com",
-  "zzom.co.uk",
-  "zzz.com",
-  "zzz.pl",
-];
-
-// Below, we have frontend form validation that prevents a user from signing up with an email address with what is likely a personal domain.
-
-function professionalEmailOnly(e) {
-  const email = document.getElementById("email").value;
-  if (filteredDomains.indexOf(email.split("@")[1]) !== -1) {
-    document.getElementById("emailError").classList.remove("invisible", "h-0");
-    document.getElementById("emailError").classList.add("pt-2");
-  } else {
-    document.getElementById("emailError").classList.add("invisible", "h-0");
-    document.getElementById("passwordError").classList.remove("pt-2");
-  }
-}
-
-// Below, we have frontend form validation that prevents a user from trying to sign up with an invalid password, enhancing security. 
-
-function passwordPass(e) {
-  const password = document.getElementById("password").value;
-  const regex = /(?=.*\d)(?=.*[a-z])(?=.*[A-Z]).{6,}/;
-  if (password.length > 4) {
-    if (!regex.test(password)) {
-      document.getElementById("passwordError").classList.remove("invisible", "h-0");
-      document.getElementById("passwordError").classList.add("pt-2");
-    } else {
-      document.getElementById("passwordError").classList.add("invisible", "h-0");
-      document.getElementById("passwordError").classList.remove("pt-2");
```

Comment:
If you look at lines 4778 - 4782, its a direct copy of lines 4763 - 4767. Can we abstract that out so we're not copy-pasting here?